huey_monitor_tests/test_app/tasks.py

```diff
@@ -129,3 +130,66 @@ def parallel_task(task, total=2000, task_num=3, **info_kwargs):
         logger.info('Start sub task no. %i', no)
         time.sleep(5)
         parallel_sub_task(parent_task_id=task.id, item_chunk=chunk, **info_kwargs)
+
+@task(context=True, retries=1)
+def sub_task_recursive(task, parent_task_id):
+    """
+    Example of implementation for recursive tasks where final number of sub-tasks is unknown.
+    Each recursive sub-task will refer to the same task as their parent task
+    """
+    logger.info('Recursive sub task started from main task: %s', parent_task_id)
+    TaskModel.objects.set_parent_task(
+        main_task_id=parent_task_id,
+        sub_task_id=task.id,
+    )
+    # let's consider we don't know yet the number of steps
+    process_info = ProcessInfo(task, desc='Recursive task execution', total=999)
```

Comment:
Hm. `total=999` is a little bit ugly, isn't it?

Can we resolve this in a other way?